source_intake/parse.py

```
@@ -0,0 +1,225 @@
+    (preserves URL-path behavior); from_text overrides to 1 (preserves
+    /api/extract wire contract).
+    """
+    cleaned = text.replace("\r", "\n")
+    cleaned = _CONTROL_CHARS_RE.sub("", cleaned)
+    cleaned = _BLANK_LINES_RE.sub("\n\n", cleaned).strip()
+
+    if len(cleaned) < min_text_length:
+        raise ParseEmpty(f"raw text {len(cleaned)} chars (min {min_text_length})")
+
+    # Title: first non-empty line if short, else host or default.
+    first_line = next((l.strip() for l in cleaned.split("\n") if l.strip()), "")
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)